### ISSUE TITLE
`unused_parens` never checks expressions behind references

### ISSUE THREAD

**Issue by lcnr** [rust-lang/rust#70084]
```rust
fn main() {
    let _ = &(5);
    // This does not warn
}
```
[playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=4f7d927326c626229ddbc94b96fc2e71)

@rustbot modify labels: +A-diagnostics, +A-macros, +T-compiler
